agent/engine/execcmd/manager_start_linux.go

```diff
@@ -46,7 +46,7 @@ func (m *manager) RestartAgentIfStopped(ctx context.Context, client dockerapi.Do
 		return NotRestarted, nil
 	}
 	res, err := m.inspectExecAgentProcess(ctx, client, execMD)
-	if err != nil {
+	if err != nil || res == nil {
```

Comment:
curious how we end up with a nil res?  I'm looking at the inspectExecAgentProcess function: https://github.com/aws/amazon-ecs-agent/blob/feature/ecs_exec/agent/engine/execcmd/manager_start_linux.go#L67

did you encounter this case?

Reply:
ended up finding a goroutine from integ test that was panicking because response was nil :)